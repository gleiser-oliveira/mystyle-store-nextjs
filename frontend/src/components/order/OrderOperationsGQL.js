import gql from 'graphql-tag';
import { perPage } from '../../config';

export const SINGLE_ORDER_QUERY = gql`
    query SINGLE_ORDER_QUERY($id: ID!) {
        order(id: $id) {
            id
            charge
            total
            createdAt
            user {
                id
            }
            products {
                id
                title
                description
                price
                image
                quantity
            }
        }
    }
`;

export const USER_ORDERS_QUERY = gql`
    query USER_ORDERS_QUERY($skip: Int = 0, $first: Int = ${perPage}) { 
        orders(first: $first, skip: $skip, orderBy: createdAt_DESC) {
            id
            total
            createdAt
            products {
                id
                title
                price
                description
                quantity
                image
            }
        }
    }
`;

export const USER_ORDERS_PAGINATION_QUERY = gql`
    query USER_ORDERS_PAGINATION_QUERY($userId: ID!) {
        ordersConnection(where: {user: {id: $userId}}) {
            aggregate {
                count
            }
        }
    }
`;
