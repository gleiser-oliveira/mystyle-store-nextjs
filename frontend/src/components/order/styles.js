import styled from 'styled-components';

export const OrderStyle = styled.div`
    background-color: white;
    max-width: 1000px;
    margin: 0 auto;
    border: 1px solid ${props => props.theme.offWhite};
    box-shadow: ${props => props.theme.boxShadow};
    padding: 2rem;
    border-top: 10px solid ${props => props.theme.mainColor};
    & > p {
        display: grid;
        grid-template-columns: 1fr 5fr;
        margin: 0;
        border-bottom: 1px solid ${props => props.theme.offWhite};
        span {
            padding: 1rem;
            &:first-child {
                font-weight: 900;
                text-align: right;
            }
        }
    }
    .order-item {
        border-bottom: 1px solid ${props => props.theme.offWhite};
        display: grid;
        grid-template-columns: 300px 1fr;
        align-items: center;
        grid-gap: 2rem;
        margin: 2rem 0;
        padding-bottom: 2rem;
        img {
            width: 100%;
            object-fit: cover;
        }
    }
`;

export const OrderListStyle = styled.div`
    display: grid;
    grid-gap: 4rem;
    grid-template-columns: repeat(auto-fit, minmax(40%, 1fr));
`;

export const OrderItemStyle = styled.li`
    background-color: white;
    box-shadow: ${props => props.theme.boxShadow};
    list-style: none;
    padding: 2rem;
    border: 1px solid ${props => props.theme.offWhite};
    h2 {
        border-bottom: 2px solid red;
        margin-top: 0;
        margin-bottom: 2rem;
        padding-bottom: 2rem;
    }

    .images {
        display: grid;
        grid-gap: 10px;
        grid-template-columns: repeat(auto-fit, minmax(0, 1fr));
        margin-top: 1rem;
        img {
            height: 200px;
            object-fit: cover;
            width: 100%;
        }
    }
    .order-meta {
        display: grid;
        grid-template-columns: repeat(auto-fit, minmax(20px, 1fr));
        display: grid;
        grid-gap: 1rem;
        text-align: center;
        & > * {
            margin: 0;
            background: rgba(0, 0, 0, 0.03);
            padding: 1rem 0;
        }
        strong {
            display: block;
            margin-bottom: 1rem;
        }
    }
`;
