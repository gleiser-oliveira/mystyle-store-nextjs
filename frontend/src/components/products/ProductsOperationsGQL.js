import gql from 'graphql-tag';
import { perPage } from '../../config';

export const ALL_PRODUCTS_QUERY = gql`
    query ALL_PRODUCTS_QUERY($skip: Int = 0, $first: Int = ${perPage}) {
        products(first: $first, skip: $skip, orderBy: createdAt_DESC) {
            id
            title
            price
            description
            image
            largeImage
            user {
                id
            }
        }
    }
`;

export const SINGLE_PRODUCT_QUERY = gql`
    query SINGLE_PRODUCT_QUERY($id: ID!) {
        product(where: { id: $id }) {
            id
            title
            description
            price
            image
            largeImage
            user {
                id
            }
        }
    }
`;

export const CREATE_PRODUCT_MUTATION = gql`
    mutation CREATE_PRODUCT_MUTATION (
        $title: String!
        $description: String!
        $price: Int!
        $image: String
        $largeImage: String
    ) {
        createProduct (
            title: $title 
            description: $description 
            price: $price
            image: $image
            largeImage: $largeImage
        ) {
            id
        }
    }
`;

export const UPDATE_PRODUCT_MUTATION = gql`
    mutation UPDATE_PRODUCT_MUTATION (
        $id: ID!
        $title: String
        $description: String
        $price: Int
    ) {
        updateProduct (
            id: $id
            title: $title 
            description: $description 
            price: $price
        ) {
            id
        }
    }
`;

export const DELETE_PRODUCT_MUTATION = gql`
    mutation DELETE_PRODUCT_MUTATION($id: ID!) {
        deleteProduct(id: $id) {
            id
        }
    }
`;

export const PRODUCTS_PAGINATION_QUERY = gql`
  query PRODUCTS_PAGINATION_QUERY {
    productsConnection {
      aggregate {
        count
      }
    }
  }
`;
