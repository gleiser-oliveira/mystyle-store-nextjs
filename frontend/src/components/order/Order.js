import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import { Query } from 'react-apollo';
import { SINGLE_ORDER_QUERY } from './OrderOperationsGQL';
import { format } from 'date-fns';
import Head from 'next/head';
import { formatPrice } from '../../lib/cartHelper';
import { ErrorMessage } from '../message/Message';
import { Center } from '../theme/generalStyles';
import { OrderStyle } from './styles';

export default class Order extends Component {
    static propTypes = {
        id: PropTypes.string.isRequired,
    };

    render() {
        return (
            <Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
                {payload => {
                    if (payload.error)
                        return <ErrorMessage error={payload.error} />;
                    if (payload.loading)
                        return (
                            <Center>
                                <Loader type="TailSpin" visible={true} />
                            </Center>
                        );
                    const order = payload.data.order;
                    return (
                        <OrderStyle data-test='order'>
                            <Head>
                                <title>Sick Fits - Order {order.id}</title>
                            </Head>
                            <p>
                                <span>Order ID:</span>
                                <span>{this.props.id}</span>
                            </p>
                            <p>
                                <span>Charge</span>
                                <span>{order.charge}</span>
                            </p>
                            <p>
                                <span>Date</span>
                                <span>{format(new Date(order.createdAt), 'MMMM d, yyyy hh:mm:ss')}</span>
                            </p>
                            <p>
                                <span>Order Total</span>
                                <span>{formatPrice(order.total)}</span>
                            </p>
                            <p>
                                <span>Item Count</span>
                                <span>{order.products.length}</span>
                            </p>
                            <div className="items">
                                {order.products.map(product => (
                                    <div className="order-item" key={product.id}>
                                        <img src={product.image} alt={product.title} />
                                        <div className="item-details">
                                            <h2>{product.title}</h2>
                                            <p>Qty: {product.quantity}</p>
                                            <p>Each: {formatPrice(product.price)}</p>
                                            <p>SubTotal: {formatPrice(product.price * product.quantity)}</p>
                                            <p>{product.description}</p>
                                        </div>
                                    </div>
                                ))}
                            </div>
                        </OrderStyle>
                    );
                }}
            </Query>
        );
    }
}
