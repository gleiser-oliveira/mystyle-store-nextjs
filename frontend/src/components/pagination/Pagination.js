import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import Head from 'next/head';
import Link from 'next/link';
import { PaginationStyle } from './styles';
import { perPage } from '../../config';

const Pagination = props => {
    return (
        <Query query={props.query} variables={props.variables}>
            {(payload) => {
                if (payload.loading)
                    // props.loading.event(props.loading.name, true);
                    return null;
                else {
                    props.loading.event(props.loading.name, false);
                    if (payload.error)
                        return <p>Error: {payload.error.message}</p>;
                    const numObjects = payload.data[props.connectionName].aggregate.count;
                    let pages = Math.ceil(numObjects / perPage);
                    if (pages === 0)
                        pages = 1;
                    const { page } = props;
                    return (
                        <PaginationStyle>
                            <Head>
                                <title>My Style | Page {page} of {pages}</title>
                            </Head>
                            <Link href={{
                                pathname: props.pathname,
                                query: { page: page - 1 },
                            }}>
                                <a className='prev' aria-disabled={page <= 1}>
                                    Prev
                                </a>
                            </Link>
                            <p>Page {page} of {pages}</p>
                            <p data-test='pagination'>{numObjects} {props.nameObjects} Total</p>
                            <Link href={{
                                pathname: props.pathname,
                                query: { page: page + 1 },
                            }}>
                                <a className='next' aria-disabled={page >= pages}>
                                    Next
                                </a>
                            </Link>
                        </PaginationStyle>
                    );
                }
            }}
        </Query>
    );
};

Pagination.propTypes = {
    query: PropTypes.object.isRequired,
    variables: PropTypes.object,
    connectionName: PropTypes.string.isRequired,
    page: PropTypes.number.isRequired,
    pathname: PropTypes.string.isRequired,
    loading: PropTypes.object.isRequired,
    nameObjects: PropTypes.string,
};

export default Pagination;
