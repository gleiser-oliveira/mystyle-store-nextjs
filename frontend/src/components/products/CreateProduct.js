import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import Router from 'next/router';
import { CREATE_PRODUCT_MUTATION, ALL_PRODUCTS_QUERY } from './ProductsOperationsGQL';
import { Form } from '../theme/generalStyles';
import { ErrorMessage } from '../message/Message';

export default class CreateProduct extends Component {
    state = {
        title: '',
        description: '',
        price: 0,
        image: '',
        largeImage: ''
    };

    handleChange = (e) => {
        const { name, type, value } = e.target;
        const val = type === 'number' ? parseFloat(value) : value;
        this.setState({ [name]: val });
    };

    handleSubmit = async(e, createProduct) => {
        e.preventDefault();
        const res = await createProduct();
        Router.push({
            pathname: '/product',
            query: { id: res.data.createProduct.id }
        });
    };

    uploadImage = async(e) => {
        const files = e.target.files;
        const data = new FormData();
        data.append('file', files[0]);
        data.append('upload_preset', 'my-style');
        const res = await fetch('https://api.cloudinary.com/v1_1/go-medias/image/upload', {
            method: 'POST',
            body: data
        });
        const file = await res.json();
        this.setState({
            image: file.secure_url,
            largeImage: file.eager[0].secure_url
        });
    };

    render() {
        return (
            <Mutation mutation={CREATE_PRODUCT_MUTATION} variables={this.state} refetchQueries={[{ query: ALL_PRODUCTS_QUERY }]}>
                {(createProduct, payload) => (
                    <Form onSubmit={async e => { this.handleSubmit(e, createProduct); }}>
                        {payload.error ? <ErrorMessage error={payload.error} /> : null}
                        <fieldset disabled={payload.loading} aria-busy={payload.loading}>
                            <label htmlFor='title'>
                                Title
                                <input
                                    type='text'
                                    id='title'
                                    name='title'
                                    placeholder='Title'
                                    required
                                    value={this.state.title}
                                    onChange={this.handleChange}/>
                            </label>
                            <label htmlFor='title'>
                                Price
                                <input
                                    type='number'
                                    id='price'
                                    name='price'
                                    placeholder='Price'
                                    required
                                    value={this.state.price}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor='title'>
                                Description
                                <textarea
                                    type='text'
                                    id='description'
                                    name='description'
                                    placeholder='Enter a Description'
                                    required
                                    value={this.state.description}
                                    onChange={this.handleChange}/>
                            </label>
                            <label htmlFor='file'>
                                Image
                                <input
                                    type='file'
                                    id='file'
                                    name='file'
                                    placeholder='Upload Image'
                                    onChange={this.uploadImage}/>
                                {this.state.image && <img width='200px' src={this.state.image} alt='Upload Preview' />}
                            </label>
                            <button type='submit'>Submit</button>
                        </fieldset>
                    </Form>
                )}
            </Mutation>
        );
    }
}
