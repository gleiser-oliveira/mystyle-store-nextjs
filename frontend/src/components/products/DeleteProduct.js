import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import { ALL_PRODUCTS_QUERY, DELETE_PRODUCT_MUTATION } from './ProductsOperationsGQL';
import { CURRENT_USER_QUERY } from '../authentication/AuthenticationOperationsGQL';

class DeleteProduct extends Component {
    updateCache = (cache, payload) => {
        // manually update the apollo cache on the client, so it matches the server
        // read cache for the products we want
        const data = cache.readQuery({ query: ALL_PRODUCTS_QUERY });
        // filter the deleted product
        data.products = data.products.filter(product => product.id !== payload.data.deleteProduct.id);
        cache.writeQuery({ query: ALL_PRODUCTS_QUERY, data });
    }

    deleteProductAction = (e, deleteProductMutation) => {
        let res = null;
        const confirmed = confirm('Are you sure you want to delete this product?');
        if (confirmed)
            res = deleteProductMutation();
        return res;
    }

    render() {
        return (
            <Mutation mutation={ DELETE_PRODUCT_MUTATION } variables={{ id: this.props.id }} refetchQueries={[{ query: CURRENT_USER_QUERY }, { query: ALL_PRODUCTS_QUERY }]} update={this.updateCache}>
                {(deleteProductMutation, payload) => (
                    // <ErrorMessage error={payload.error}></ErrorMessage>
                    <button onClick={ e => this.deleteProductAction(e, deleteProductMutation)}>
                        {this.props.children}
                    </button>
                )}
            </Mutation>
        );
    }
}

DeleteProduct.propTypes = {
    id: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
};

export default DeleteProduct;
