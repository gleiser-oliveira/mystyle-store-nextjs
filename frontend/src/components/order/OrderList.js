import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { USER_ORDERS_QUERY, USER_ORDERS_PAGINATION_QUERY } from './OrderOperationsGQL';
import Link from 'next/link';
import { formatDistance } from 'date-fns';
import Pagination from '../pagination/Pagination';
import Loader from 'react-loader-spinner';
import { formatPrice } from '../../lib/cartHelper';
import { ErrorMessage } from '../message/Message';
import { Center } from '../theme/generalStyles';
import { OrderItemStyle, OrderListStyle } from './styles';
import { perPage } from '../../config';

class OrderList extends Component {
    state = {
        pagLoading1: false,
        pagLoading2: false,
    };

    setLoading = (pag, isLoading) => {
        if (this.state[pag] !== isLoading)
            this.setState({ [pag]: isLoading });
    };

    renderImgGrid(products) {
        let grid = products;
        let overflow = null;
        if (products.length > 7) {
            grid = products.slice(0, 7);
            overflow = <div className="order-meta"><span>...</span></div>;
        }

        const imgs = grid.map(product => <img key={product.id} src={product.image} alt={product.title} />);

        return (
            <>
                {imgs}
                {overflow}
            </>
        );
    };

    render() {
        return (
            <>
                <Center>
                    <Pagination
                        page={this.props.page}
                        loading={{ event: this.setLoading, name: 'pagLoading1' }}
                        query={USER_ORDERS_PAGINATION_QUERY}
                        variables={{ userId: this.props.userData.id }}
                        nameObjects='Orders'
                        pathname='orders'
                        connectionName='ordersConnection'
                    />
                </Center>
                <Query query={USER_ORDERS_QUERY} variables={{
                    skip: this.props.page * perPage - perPage,
                    first: perPage
                }}>
                    {(payload) => {
                        if (payload.loading || this.state.pagLoading1 || this.state.pagLoading2)
                            return (
                                <Center>
                                    <Loader type="TailSpin" visible={true} />
                                </Center>
                            );
                        if (payload.error)
                            return (
                                <Center>
                                    <ErrorMessage error={payload.error} />
                                </Center>
                            );
                        return (
                            <Center>
                                <OrderListStyle>
                                    {payload.data.orders.length < 1 && <Center><p>No orders were found.</p></Center>}
                                    {payload.data.orders.map(order => (
                                        <OrderItemStyle key={order.id}>
                                            <Link
                                                href={{
                                                    pathname: '/order',
                                                    query: { id: order.id },
                                                }}
                                            >
                                                <a>
                                                    <div className="order-meta">
                                                        <p>{order.products.reduce((a, b) => a + b.quantity, 0)} Items</p>
                                                        <p>{order.products.length} Products</p>
                                                        <p>{formatDistance(new Date(order.createdAt), new Date())}</p>
                                                        <p>{formatPrice(order.total)}</p>
                                                    </div>
                                                    <div className="images">
                                                        {this.renderImgGrid(order.products)}
                                                    </div>
                                                </a>
                                            </Link>
                                        </OrderItemStyle>
                                    ))}
                                </OrderListStyle>
                            </Center>
                        );
                    }}
                </Query>
                <Center>
                    <Pagination
                        page={this.props.page}
                        loading={{ event: this.setLoading, name: 'pagLoading2' }}
                        query={USER_ORDERS_PAGINATION_QUERY}
                        variables={{ userId: this.props.userData.id }}
                        nameObjects='Orders'
                        pathname='orders'
                        connectionName='ordersConnection'
                    />
                </Center>
            </>
        );
    }
}

OrderList.propTypes = {
    page: PropTypes.number,
    userData: PropTypes.shape({
        id: PropTypes.string,
    }),
};

export default OrderList;
