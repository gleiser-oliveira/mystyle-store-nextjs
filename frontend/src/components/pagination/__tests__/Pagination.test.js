import { mount } from 'enzyme';
import Pagination from '../Pagination';
import toJSON from 'enzyme-to-json';
import { USER_ORDERS_PAGINATION_QUERY } from '../../order/OrderOperationsGQL';
import { PRODUCTS_PAGINATION_QUERY } from '../../products/ProductsOperationsGQL';
import { MockedProvider } from '@apollo/react-testing';
import { updateWrapper } from '../../../lib/testUtils';

const pagLoadingMock = {
    event: jest.fn(),
    name: 'mock',
};

const makeMocksFor = (query, connectionName, count) => {
    return [
        {
            request: { query: query },
            result: {
                data: {
                    [connectionName]: {
                        __typename: 'aggregate',
                        aggregate: {
                            __typename: 'count',
                            count: count,
                        },
                    },
                },
            },
        },
    ];
};

describe('testing the <Pagination/> component', () => {
    it('renders pagination for 37 products', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(PRODUCTS_PAGINATION_QUERY, 'productsConnection', 37)}>
                <Pagination loading={pagLoadingMock} page={1} query={PRODUCTS_PAGINATION_QUERY} nameObjects='Products' pathname='products' connectionName='productsConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const pagination = wrapper.find('Pagination');

        expect(toJSON(pagination)).toMatchSnapshot();
    });
    it('renders pagination for 83 orders', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(USER_ORDERS_PAGINATION_QUERY, 'ordersConnection', 83)}>
                <Pagination loading={pagLoadingMock} page={1} query={USER_ORDERS_PAGINATION_QUERY} nameObjects='Orders' pathname='orders' connectionName='ordersConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const pagination = wrapper.find('Pagination');
        expect(toJSON(pagination)).toMatchSnapshot();
    });
    it('disables only the previous button on the first page', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(PRODUCTS_PAGINATION_QUERY, 'productsConnection', 8)}>
                <Pagination loading={pagLoadingMock} page={1} query={PRODUCTS_PAGINATION_QUERY} nameObjects='Products' pathname='products' connectionName='productsConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const prevBtn = wrapper.find('a.prev');
        const nextBtn = wrapper.find('a.next');
        expect(prevBtn.prop('aria-disabled')).toEqual(true);
        expect(nextBtn.prop('aria-disabled')).toEqual(false);
    });
    it('disables only the next button on the last page', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(PRODUCTS_PAGINATION_QUERY, 'productsConnection', 8)}>
                <Pagination loading={pagLoadingMock} page={2} query={PRODUCTS_PAGINATION_QUERY} nameObjects='Products' pathname='products' connectionName='productsConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const prevBtn = wrapper.find('a.prev');
        const nextBtn = wrapper.find('a.next');
        expect(prevBtn.prop('aria-disabled')).toEqual(false);
        expect(nextBtn.prop('aria-disabled')).toEqual(true);
    });
    it('enables both buttons on a middle page', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(USER_ORDERS_PAGINATION_QUERY, 'ordersConnection', 15)}>
                <Pagination loading={pagLoadingMock} page={2} query={USER_ORDERS_PAGINATION_QUERY} nameObjects='Orders' pathname='orders' connectionName='ordersConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const prevBtn = wrapper.find('a.prev');
        const nextBtn = wrapper.find('a.next');
        expect(prevBtn.prop('aria-disabled')).toEqual(false);
        expect(nextBtn.prop('aria-disabled')).toEqual(false);
    });
    it('disbles both buttons in a single page', async() => {
        const wrapper = mount(
            <MockedProvider mocks={makeMocksFor(USER_ORDERS_PAGINATION_QUERY, 'ordersConnection', 1)}>
                <Pagination loading={pagLoadingMock} page={1} query={USER_ORDERS_PAGINATION_QUERY} nameObjects='Orders' pathname='orders' connectionName='ordersConnection'/>
            </MockedProvider>
        );

        await updateWrapper(wrapper);

        const prevBtn = wrapper.find('a.prev');
        const nextBtn = wrapper.find('a.next');
        expect(prevBtn.prop('aria-disabled')).toEqual(true);
        expect(nextBtn.prop('aria-disabled')).toEqual(true);
    });
});
