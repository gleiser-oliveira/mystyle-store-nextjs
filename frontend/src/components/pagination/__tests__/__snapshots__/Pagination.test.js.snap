// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <Pagination/> component renders pagination for 37 products 1`] = `
<Pagination
  connectionName="productsConnection"
  loading={
    Object {
      "event": [MockFunction] {
        "calls": Array [
          Array [
            "mock",
            false,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "name": "mock",
    }
  }
  nameObjects="Products"
  page={1}
  pathname="products"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "PRODUCTS_PAGINATION_QUERY",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "productsConnection",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "aggregate",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "count",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 112,
        "start": 0,
      },
    }
  }
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "PRODUCTS_PAGINATION_QUERY",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "productsConnection",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "aggregate",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "count",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 112,
          "start": 0,
        },
      }
    }
  >
    <styles__PaginationStyle>
      <div
        className="styles__PaginationStyle-sc-45u64h-0 fSgPSQ"
      >
        <Head>
          <Component
            handleStateChange={null}
            inAmpMode={false}
            reduceComponentsToState={[Function]}
          />
        </Head>
        <Link
          href={
            Object {
              "pathname": "products",
              "query": Object {
                "page": 0,
              },
            }
          }
        >
          <a
            aria-disabled={true}
            className="prev"
            href="products?page=0"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Prev
          </a>
        </Link>
        <p>
          Page 
          1
           of 
          10
        </p>
        <p
          data-test="pagination"
        >
          37
           
          Products
           Total
        </p>
        <Link
          href={
            Object {
              "pathname": "products",
              "query": Object {
                "page": 2,
              },
            }
          }
        >
          <a
            aria-disabled={false}
            className="next"
            href="products?page=2"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Next
          </a>
        </Link>
      </div>
    </styles__PaginationStyle>
  </Query>
</Pagination>
`;

exports[`testing the <Pagination/> component renders pagination for 83 orders 1`] = `
<Pagination
  connectionName="ordersConnection"
  loading={
    Object {
      "event": [MockFunction] {
        "calls": Array [
          Array [
            "mock",
            false,
          ],
          Array [
            "mock",
            false,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "name": "mock",
    }
  }
  nameObjects="Orders"
  page={1}
  pathname="orders"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "USER_ORDERS_PAGINATION_QUERY",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "where",
                    },
                    "value": Object {
                      "fields": Array [
                        Object {
                          "kind": "ObjectField",
                          "name": Object {
                            "kind": "Name",
                            "value": "user",
                          },
                          "value": Object {
                            "fields": Array [
                              Object {
                                "kind": "ObjectField",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "userId",
                                  },
                                },
                              },
                            ],
                            "kind": "ObjectValue",
                          },
                        },
                      ],
                      "kind": "ObjectValue",
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "ordersConnection",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "aggregate",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "count",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "userId",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 189,
        "start": 0,
      },
    }
  }
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "USER_ORDERS_PAGINATION_QUERY",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "where",
                      },
                      "value": Object {
                        "fields": Array [
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "user",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "userId",
                                    },
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "kind": "ObjectValue",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "ordersConnection",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "aggregate",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "count",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "userId",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 189,
          "start": 0,
        },
      }
    }
  >
    <styles__PaginationStyle>
      <div
        className="styles__PaginationStyle-sc-45u64h-0 fSgPSQ"
      >
        <Head>
          <Component
            handleStateChange={null}
            inAmpMode={false}
            reduceComponentsToState={[Function]}
          />
        </Head>
        <Link
          href={
            Object {
              "pathname": "orders",
              "query": Object {
                "page": 0,
              },
            }
          }
        >
          <a
            aria-disabled={true}
            className="prev"
            href="orders?page=0"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Prev
          </a>
        </Link>
        <p>
          Page 
          1
           of 
          21
        </p>
        <p
          data-test="pagination"
        >
          83
           
          Orders
           Total
        </p>
        <Link
          href={
            Object {
              "pathname": "orders",
              "query": Object {
                "page": 2,
              },
            }
          }
        >
          <a
            aria-disabled={false}
            className="next"
            href="orders?page=2"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Next
          </a>
        </Link>
      </div>
    </styles__PaginationStyle>
  </Query>
</Pagination>
`;
